src/coreclr/System.Private.CoreLib/src/System/Threading/SynchronizedMethodHelpers.CoreCLR.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+
+using Debug = System.Diagnostics.Debug;
+
+namespace System.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// Set of helpers used to implement synchronized methods.
+    /// </summary>
+    internal static class SynchronizedMethodHelpers
+    {
+        private static void MonitorEnter(object obj, ref bool lockTaken)
```

Comment:
We have been moving away from the native AOT factoring with JIT helpers on the side under `Internal\Runtime\CompilerHelpers\*`, and placing the JIT helpers into their natural homes instead. For example, we used to have `Internal\Runtime\CompilerHelpers\MathHelpers.cs` that is gone now and merged into System.Math. It is better for CoreCLR (fewer types loaded) and the code is easier to understand when everything related is in the same file.

I would move Monitor.Enter/Exit helpers to Monitor, both here and in the NativeAOT version.

Reply:
(Give them distinct name like `SynchronizedMethodEnter/Exit` so that there are no collisions with the regular Enter/Exit.)